File: c/magic/magic.c
//
// Copyright 2014 YelloSoft. All rights reserved.

// Redistribution and use in source and binary forms, with or without modification, are permitted
// provided that the following conditions are met:

// Redistributions of source code must retain the above copyright notice, this list of conditions
// and the following disclaimer.
// Redistributions in binary form must reproduce the above copyright notice, this list of
// conditions and the following disclaimer in the documentation and/or other materials provided
// with the distribution.
// THIS SOFTWARE IS PROVIDED BY THE FREEBSD PROJECT ``AS IS'' AND ANY EXPRESS OR IMPLIED
// WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
// FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE FREEBSD PROJECT OR CONTRIBUTORS BE
// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
// (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA,
// OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
// THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.

// The views and conclusions contained in the software and documentation are those of the authors
// and should not be interpreted as representing official policies, either expressed or implied, of
// YelloSoft.
//

#include <stdio.h>

int main() {
  unsigned int magic[16] = {
    0x09, 0xf9, 0x11, 0x02, 0x9d, 0x74, 0xe3, 0x5b,
    0xd8, 0x41, 0x56, 0xc5, 0x63, 0x56, 0x88, 0xbc
  };

  int i;
  for (i=0; i<7; i++) {
    magic[15]++;

    if (i!=3) {
      printf(
        "%02x %02x %02x %02x %02x %02x %02x %02x %02x %02x %02x %02x %02x %02x %02x %02x\n",
        magic[0], magic[1], magic[2], magic[3], magic[4], magic[5], magic[6], magic[7],
        magic[8], magic[9], magic[10], magic[11], magic[12], magic[13], magic[14], magic[15]
      );
    }
    else {
      printf("                 [ redacted ]\n");
    }
  }

  return 0;
}

File: scala/Magic.scala
#!/usr/bin/env scala

object Magic {
  val NUMBERS = List(
    0xbd, 0x88, 0x56, 0x63,
    0xc5, 0x56, 0x41, 0xd8,
    0x5b, 0xe3, 0x74, 0x9d,
    0x02, 0x11, 0xf9, 0x09
  )

  def magic : String = {
    def magicN(n : Int) : String = n match {
      case 3 => " " * 17 + "[ redacted ]"
      case _ => {
        var pieces = ""

        ((NUMBERS.head + n) :: NUMBERS.tail).reverse.map((byte : Int) => String.format("%02x", new Integer(byte))).foreach { piece => pieces += piece + " " }

        pieces
      }
    }

    var lines = ""

    (0 to 6).map(magicN).foreach { line => lines += line + "\n" }

    lines
  }

  def main(args : Array[String]) : Unit = printf(magic)
}
